plan/logical_plan_builder.go

```diff
@@ -1212,11 +1214,29 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 			pkCol = schema.Columns[schema.Len()-1]
 		}
 	}
+	p.SetSchema(schema)
+
+	// for calculating generated column
+	p.GenValues = make(map[int]expression.Expression)
+	for idx, column := range columns {
+		if column.IsGenerated() && !column.GeneratedStored {
+			expr, _, err := b.rewrite(column.GeneratedExpr, p, nil, true)
+			if err != nil {
+				b.err = errors.Trace(err)
+				return nil
+			}
+			if p.GenValues == nil {
```

Comment:
I want the `GenValues` is not nil only if there are really some generated columns in the table. So we can save some space. I removed line `1220`.